tests/integration_tests/document_loaders/test_mongodb.py

```diff
@@ -0,0 +1,121 @@
+import os
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from motor.motor_asyncio import AsyncIOMotorClient
+
+from langchain.docstore.document import Document
+from langchain.document_loaders import MongodbLoader
+
+if "MONGODB_API_KEY" in os.environ:
+    mongo_api_key_set = True
+    mongo_api_key = os.environ["MONGODB_API_KEY"]
+else:
+    mongo_api_key_set = False
+
+
+@pytest.fixture
+def docs():
+    return [
+        {"_id": "1", "address": {"building": "1", "room": "1"}},
+        {"_id": "2", "address": {"building": "2", "room": "2"}},
+    ]
+
+
+@pytest.fixture
+def expected_documents():
+    return [
+        Document(
+            page_content="{'_id': '1', 'address': {'building': '1', 'room': '1'}}",
+            metadata={"database": "sample_restaurants", "collection": "restaurants"},
+        ),
+        Document(
+            page_content="{'_id': '2', 'address': {'building': '2', 'room': '2'}}",
+            metadata={"database": "sample_restaurants", "collection": "restaurants"},
+        ),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_async_load_mocked(mocker, docs, expected_documents):
+    async def mock_find():
+        for doc in docs:
+            yield doc
+
+    mock_collection = MagicMock()
+    mock_db = MagicMock()
+    mock_client = MagicMock()
+
+    mocker.patch.object(mock_collection, "find", return_value=mock_find())
+    mocker.patch.object(AsyncIOMotorClient, "get_database", return_value=mock_db)
+    mocker.patch.object(mock_db, "get_collection", return_value=mock_collection)
+
+    with patch("motor.motor_asyncio.AsyncIOMotorClient", return_value=mock_client):
+        loader = MongodbLoader(
+            "mongodb://localhost:27017", "sample_restaurants", "restaurants"
+        )
+
+    documents = await loader._async_load()
+
+    assert documents == expected_documents
+
+
+@pytest.mark.asyncio
+async def test_async_partial_load_mocked(mocker, docs, expected_documents):
+    async def mock_find():
+        for doc in docs:
+            yield doc
+
+    expected_documents.remove(expected_documents[1])
+
+    mock_collection = MagicMock()
+    mock_db = MagicMock()
+    mock_client = MagicMock()
```

Comment:
this is great! :) Usually we use mocks in unittest, but it’s not available here as unittest in langchain doesn’t have all dependent libraries imported… So I think integration test is the best option now.